runtime/io.c

```diff
@@ -736,12 +740,14 @@ CAMLprim value caml_ml_set_binary_mode(value vchannel, value mode)
     caml_sys_error(NO_ARG);
   }
 #endif
+  Lock(channel);
```

Comment:
The WIN32 only access to `channel->flags` above is a potential data race, that lock should be moved before the `ifdef` section.

Reply:
My bad; fixed.